src/main/kotlin/org/rust/ide/annotator/fixes/ChangeReprAttributeFix.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator.fixes
+
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.cargo.project.workspace.PackageOrigin
+import org.rust.ide.presentation.render
+import org.rust.lang.core.psi.RsEnumItem
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.types.ty.Ty
+import org.rust.lang.core.types.ty.isIntegral
+
+class ChangeReprAttributeFix(
+    element: RsElement,
+    private val actualTy: Ty
+) : LocalQuickFixAndIntentionActionOnPsiElement(element) {
+    private val _text: String = run {
+        val owner = findEnumOwner(startElement) ?: error("enum not found")
+
+        val enumName = owner.nameIdentifier!!.text
+
+        // actualTy will be some numeric type, therefore rendering is simple
+        val rendered = actualTy.render()
+
+        "Change representation of enum '$enumName' to '#[repr($rendered)]'"
+    }
+
+    override fun getText(): String = _text
+    override fun getFamilyName(): String = "Change repr attribute"
+
+    override fun invoke(
+        project: Project,
+        file: PsiFile,
+        editor: Editor?,
+        startElement: PsiElement,
+        endElement: PsiElement
+    ) {
+        val owner = (findEnumOwner(startElement) ?: return) as RsDocAndAttributeOwner
+
+        val reprAttributes = owner.queryAttributes.reprAttributes.toList()
+        val newOuterAttribute = RsPsiFactory(project).createOuterAttr("repr(${actualTy.render()})")
+
+        when (reprAttributes.size) {
+            0 -> {
+                // create the repr attribute
+                owner.addBefore(newOuterAttribute, owner.firstChild)
+            }
+            1 -> {
+                // replace the repr attribute
+                val reprOuterAttribute = reprAttributes[0].parent
+                reprOuterAttribute.replace(newOuterAttribute)
+            }
+
+            // multiple #[repr(...)] attributes are disallowed by "conflicting_repr_hints" hard lint
+            else -> return
+        }
+    }
+
+    companion object {
+        private fun findEnumOwner(element: PsiElement): RsEnumItem? = element.contextStrict()
+
+        fun createIfCompatible(element: RsElement, actualTy: Ty): ChangeReprAttributeFix? {
+            if (element.containingCrate?.origin != PackageOrigin.WORKSPACE) return null
+
+            // fix only works for integral expressions inside enums. expression type should not be an alias
+            if (actualTy.isIntegral && actualTy.aliasedBy == null && findEnumOwner(element) != null)
```

Comment:
`findEnumOwner(element) != null` produces a false-positive in this case:

<img width="812" alt="Screen Shot 2021-12-01 at 06 09 04" src="https://user-images.githubusercontent.com/6079006/144164941-316b8bba-2b61-4d3e-b398-008c48d8790d.png">